c10/core/DispatchKeySet.h

```diff
@@ -8,6 +7,28 @@
 
 namespace c10 {
 
+struct FunctionalityOffsetAndMask {
+    // empty constructor shouldn't be used; only needed to initialize
+    // the array before populating it.
+    FunctionalityOffsetAndMask() {}
+    FunctionalityOffsetAndMask(uint16_t offset, uint16_t mask) :
+        offset(offset), mask(mask) {}
+    // This needs to big enough to cover the size of the operator table.
```

Comment:
I should definitely assert for the mask (the `uint16_t` mask means we only support <= 16 backend bits).

I didn't bother asserting for the offset though, because it stores an index, and not a bit pattern. So the `uint16_t` offset will be fine as long as the runtime operator table doesn't have more than `2^16=65k` entries, which... we should probably realistically never hit?

Reply:
I would agree but the static assert is free and that could catch some random bug that makes the operator table way bigger than it should be?